src/GitVersionCore/GitVersionCacheKeyFactory.cs

```diff
@@ -23,10 +24,106 @@ private static string GetGitSystemHash(GitPreparer gitPreparer, IFileSystem file
         {
             var dotGitDirectory = gitPreparer.GetDotGitDirectory();
 
-            // Maybe using timestamp in .git/refs directory is enough?
-            var lastGitRefsChangedTicks = fileSystem.GetLastDirectoryWrite(Path.Combine(dotGitDirectory, "refs"));
+            // traverse the directory and get a list of files, use that for GetHash
+            var traverse = TraverseTree(Path.Combine(dotGitDirectory, "refs"));
 
-            return GetHash(dotGitDirectory, lastGitRefsChangedTicks.ToString());
+            return GetHash(traverse.ToArray());
+        }
+
+        // lifted from https://msdn.microsoft.com/en-us/library/bb513869.aspx
+        public static List<string> TraverseTree(string root)
+        {
+            var result = new List<string>();
+            result.Add(root);
+            // Data structure to hold names of subfolders to be
+            // examined for files.
+            Stack<string> dirs = new Stack<string>(20);
```

Comment:
TraverseTree comes from https://msdn.microsoft.com/en-us/library/bb513869.aspx - I didn't to change it too much - but if you'd like me to...